lib/plugin/variable-collector.js

```diff
@@ -43,15 +47,46 @@ VariableCollector.prototype = {
 	getVariables: function(aImports) {
 		const mVariables = {};
 
+		const aPosixImports = aImports.map(toPosix);
+
+		function shouldIncludeVariable({rootFilename, filename}) {
+			// Ensure posix paths
+			rootFilename = toPosix(rootFilename);
+			filename = toPosix(filename);
+
+			// Simple case where variable is defined in root file
+			if (rootFilename === filename) {
+				return true;
+			}
+
+			const dirname = posixPath.dirname(filename);
+			const baseFileName = posixPath.basename(rootFilename); // usually library.source.less
+
+			function check(currentDirname) {
+				currentDirname = posixPath.normalize(currentDirname);
+				if (currentDirname === "." || currentDirname === "/") {
+					// Check for root (relative / absolute)
+					return false;
+				}
+				const libraryBaseFile = posixPath.join(currentDirname, baseFileName);
+				if (libraryBaseFile === rootFilename || aPosixImports.includes(libraryBaseFile)) {
+					return true;
+				} else {
+					// Check whether libraryBaseFile of parent directory is imported
+					return check(posixPath.join(currentDirname, ".."));
```

Comment:
path.dirname

Reply:
And maybe exit if `posixPath.dirname(currentDirname) === currentDirname`?

Reply:
Done